pkg/karmadactl/cmdinit/cert/cert_test.go

```diff
@@ -106,7 +123,68 @@ func TestGenCerts(t *testing.T) {
 	if err := GenCerts(TestCertsTmp, "", "", etcdServerCertConfig, etcdClientCertCfg, karmadaCertCfg, apiserverCertCfg, frontProxyClientCertCfg); err != nil {
 		t.Fatal(err)
 	}
-	if err := GenCerts(TestCertsTmp, TestCaCertPath, TestCaKeyPath, etcdServerCertConfig, etcdClientCertCfg, karmadaCertCfg, apiserverCertCfg, frontProxyClientCertCfg); err != nil {
-		fmt.Println(err)
+	if err := checkCertFiles(TestCertsTmp, certFiles); err != nil {
+		t.Fatal(err)
+	} else {
+		klog.Infof("All certificate files are present without CA certificates address parameter exists")
+	}
+
+	if err := GenCerts(TestCertsTmpWithArg, TestCaCertPath, TestCaKeyPath, etcdServerCertConfig, etcdClientCertCfg, karmadaCertCfg, apiserverCertCfg, frontProxyClientCertCfg); err != nil {
+		t.Fatal(err)
+	}
+	if err := checkCertFiles(TestCertsTmpWithArg, certFiles); err != nil {
+		t.Fatal(err)
+	} else {
+		klog.Infof("All certificate files are present with CA certificates address parameter exists")
+	}
+
+	if ok, err := compareCertFilesInDirs(TestCertsTmp, TestCertsTmpWithArg, "ca.crt"); !ok || err != nil {
```

Comment:
What should I do to solve the problem here? For example, I designed separate subfolders for two GenCerts in one folder, or I misunderstood your meaning

I consider adding the function of the comparison file to determine whether the ca certificate before and after is the same, maybe I have a design error here, please give further modification plan

Reply:
You're right,  I misunderstood your meaning.  `compareCertFilesInDirs` is working for compare the files from two test case.

I think this is not necessary, WDYT  @chaosi-zju  @zhzhuang-zju

Reply:
refer to https://github.com/karmada-io/karmada/blob/a87ec2a61abadf9f102440b1f64cf338b2ee5518/pkg/karmadactl/cmdinit/cert/cert.go#L314-L333
When the ca exists, the existing ca is used, otherwise the ca is regenerated. I think it is necessary to compare the two ca files.
First，`GenCerts(TestCertsTmp, "", "", etcdServerCertConfig, etcdClientCertCfg, karmadaCertCfg, apiserverCertCfg, frontProxyClientCertCfg)` will generate files `./test-certs-tmp-without-ca-certificate/ca.crt` and `./test-certs-tmp-without-ca-certificate/ca.key`，and then `GenCerts(TestCertsTmpWithArg, TestCaCertPath, TestCaKeyPath, etcdServerCertConfig, etcdClientCertCfg, karmadaCertCfg, apiserverCertCfg, frontProxyClientCertCfg)` will use them instead of regenerating new ones. Comparing the two files verifies exactly this scenario

Reply:
Thanks for your clarify.

/lgtm